testsuite/cluster-test/src/experiments/cpu_flamegraph.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::experiments::ExperimentParam;
+use crate::{
+    cluster::Cluster, experiments::Context, experiments::Experiment, instance, instance::Instance,
+};
+use futures::future::{BoxFuture, FutureExt};
+
+use crate::effects::{Action, GenerateCpuFlamegraph};
+use std::{
+    collections::HashSet,
+    fmt::{Display, Error, Formatter},
+    time::Duration,
+};
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+pub struct CpuFlamegraphParams {
+    #[structopt(
+        long,
+        default_value = "60",
+        help = "Number of seconds for which perf should be run"
+    )]
+    pub duration_secs: usize,
+}
+
+pub struct CpuFlamegraph {
+    duration_secs: usize,
+    perf_instance: Instance,
+}
+
+impl ExperimentParam for CpuFlamegraphParams {
+    type E = CpuFlamegraph;
+    fn build(self, cluster: &Cluster) -> Self::E {
+        let (perf_instance, _) = cluster.split_n_validators_random(1);
+        let perf_instance = perf_instance.get_all_instances()[0].clone();
+        Self::E {
+            duration_secs: self.duration_secs,
+            perf_instance,
+        }
+    }
+}
+
+impl Experiment for CpuFlamegraph {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&[self.perf_instance.clone()])
+    }
+
+    fn run<'a>(
+        &'a mut self,
+        context: &'a mut Context,
+    ) -> BoxFuture<'a, anyhow::Result<Option<String>>> {
+        async move {
+            let buffer = Duration::from_secs(60);
+            let mut tx_emitter = context.tx_emitter.clone();
```

Comment:
This can be rewritten in one task, so that you don't need to clone tx_emitter. Something like that:
```
let emit_future = tx_emitter.emit_txn_for(tx_emitter_duration, validator_instances);
let flame_graph_future = GenerateCpuFlamegraph::new(self.perf_instance.clone(), self.duration_secs).apply();
let flame_graph_future = tokio::time::delay_for(buffer).then(flame_graph_future);
join_all(emit_future, flame_graph_future).await?
```
Might need some `.map` to make sure both futures return same type, which is required by `join_all`